packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -432,26 +430,23 @@ function hydrateChildren(
     // found in this case.
     expectAddlSiblings: boolean
 ) {
-    let hasWarned = false;
+    let mismatchedChildren = false;
     let nextNode: Node | null = node;
     const { renderer } = owner;
+    const { getChildNodes } = renderer;
+
+    const serverNodes =
+        process.env.NODE_ENV !== 'production'
+            ? Array.from(getChildNodes(parentNode)).map((c) => c.cloneNode(true))
```

Comment:
`ArrayFrom`, `ArrayMap`, `renderer.cloneNode`, etc. Also the `ArrayMap` is not necessary because the [second argument](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from#parameters) to `ArrayFrom` can be a map function.